### PR TITLE
feat(components): [el-menu] add jump external link

### PR DESCRIPTION
**Pull request by xiaoxian521**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/menu/src/menu.ts [hunk 1]
```diff
@@ -167,6 +168,11 @@ export default defineComponent({
       const { index, indexPath } = menuItem
       if (index === undefined || indexPath === undefined) return
 
+      if (index.includes('http')) {
```
Comment (JeremyWuuuuu) [inline packages/components/menu/src/menu.ts:171]: **Suggestion:**

`index.startsWith`

File: packages/components/menu/src/menu.ts [hunk 2]
```diff
@@ -10,10 +10,11 @@ import {
   withDirectives,
   reactive,
 } from 'vue'
+import split from 'lodash/split'
```
Comment (sxzz) [inline packages/components/menu/src/menu.ts:13]: We can use `String.prototype.split`

File: packages/components/menu/src/menu.ts [hunk 3]
```diff
@@ -167,6 +168,11 @@ export default defineComponent({
       const { index, indexPath } = menuItem
       if (index === undefined || indexPath === undefined) return
 
+      if (index.includes('http')) {
+        createLink(`http${split(index, 'http')[1]}`)
+        return
+      }
```
Comment (sxzz) [inline packages/components/menu/src/menu.ts:174]:
```suggestion
      if (index.startsWith('http')) {
        createLink(index)
        return
      }
```

File: packages/components/menu/src/menu.ts (hunk 2 above)
Comment (JeremyWuuuuu) [inline packages/components/menu/src/menu.ts:13]: Remove this.

Comment (JeremyWuuuuu) [reply packages/components/menu/src/menu.ts:13]: Prefer using `String.prototype.split`

File: packages/utils/util.ts [hunk 4]
```diff
@@ -257,3 +257,14 @@ export const refAttacher = <T extends HTMLElement | ComponentPublicInstance>(
     ref.value = val
   }
 }
+
+export function createLink(link: string) {
+  const href: HTMLElement = document.createElement('a')
+  const dom: HTMLElement | null = document.getElementById('el-external')
+  href.setAttribute('href', link)
+  href.setAttribute('target', '_blank')
```
Comment (JeremyWuuuuu) [inline packages/utils/util.ts:265]: For security, please add 
```ts
href.setAttritbute('rel', 'noreferrer noopener')
```

File: packages/utils/util.ts [hunk 5]
```diff
@@ -257,3 +257,14 @@ export const refAttacher = <T extends HTMLElement | ComponentPublicInstance>(
...
+  href.setAttribute('id', 'el-external')
+  dom && document.body.removeChild(dom)
+  document.body.appendChild(href)
+  href.click()
+}
```
Comment (sxzz) [inline packages/utils/util.ts:270]: Please destroy link element after using `href.remove()`

File: packages/utils/util.ts [hunk 6]
```diff
@@ -257,3 +257,14 @@ export const refAttacher = <T extends HTMLElement | ComponentPublicInstance>(
     ref.value = val
   }
 }
+
+export function createLink(link: string) {
+  const href: HTMLElement = document.createElement('a')
```
Comment (JeremyWuuuuu) [inline packages/utils/util.ts:262]: Prefer `link` or `$a` for the variable name, href represents `hyper reference` which does not meet the context here.

File: packages/utils/util.ts [hunk 7]
```diff
@@ -257,3 +257,14 @@ export const refAttacher = <T extends HTMLElement | ComponentPublicInstance>(
     ref.value = val
   }
 }
+
+export function createLink(link: string) {
```
Comment (sxzz) [inline packages/utils/util.ts:261]: Suggestion: rename to `openLink`

```suggestion
export function openLink(link: string) {
```

**Comment by xiaoxian521** [comment]
After investigation, this function can be implemented in the beforeEach routing hook

**Pull request closed without merging**